index.js

```diff
@@ -12,11 +12,7 @@ const defaultCacheKey = (...arguments_) => {
 
 	if (arguments_.length === 1) {
 		const [firstArgument] = arguments_;
-		if (
-			firstArgument === null ||
-			firstArgument === undefined ||
-			(typeof firstArgument !== 'function' && typeof firstArgument !== 'object')
-		) {
+		if (typeof firstArgument !== 'object' || firstArgument === null) {
```

Comment:
It took me a while to realize what this code does. think this would make it more readable:

```suggestion
		const isObject = typeof firstArgument === 'object' && firstArgument !== null;
		const isPrimitive = !isObject;
		if (isPrimitive) {
```

Reply:
Now I'm not sure that we need the extra `isPrimitive` step. `!isObject` makes enough sense on its own.

Reply:
I disagree. It's not obvious to everyone that `!isObject` means it's a primitive. Doesn't hurt to make it perfectly clear.

Reply:
I see that `primitive` is also mentioned in the readme, so it sorta makes sense to clarify it here as well.